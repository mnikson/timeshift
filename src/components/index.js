import Header from './Header';
import Loading from './Loading';
import LocationInfo from './LocationInfo';
import PageLayout from './PageLayout';

export {
  Header,
  Loading,
  LocationInfo,
  PageLayout
};
