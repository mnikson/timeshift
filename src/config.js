require('dotenv').config();

const apiUrl = process.env.REACT_APP_API_SERVER_URL || 'http://localhost:4000';

module.exports = {
  /**
   * Root DOM element
   */
  rootElement: 'root',

  /**
   * Application name
   */
  applicationName: process.env.REACT_APP_APPLICATION_NAME || 'Timeshift',
  /**
   * API url
   */
  apiUrl
};
