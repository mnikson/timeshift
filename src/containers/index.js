/**
 * Containers main file
 */

import LocationsContainer from './LocationsContainer';

export {
  LocationsContainer
};
