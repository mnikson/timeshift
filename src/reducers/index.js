/**
 * Reducers index file
 */

// Libs
import { combineReducers } from 'redux';

// Reducers
import locations from './locations';

export default combineReducers({
  locations
});
