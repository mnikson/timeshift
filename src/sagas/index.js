/**
 * @file sagas
 * @description Sagas main fine
 */

// Libs
import { all, fork } from 'redux-saga/effects';

// Sagas
import location from './location';

export default function* rootSaga() {
  yield all([
    location
  ].map(fork));
}
