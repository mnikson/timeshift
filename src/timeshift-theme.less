@import url('https://fonts.googleapis.com/css2?family=Lato');

@font-name: 'Lato';

@primary-color: #1b1919;
@font-family: @font-name;
@primary-background-color: #ffffff;

// Typography
@text-color: #001122;

// Locations list
@location-item-bg: RGBA(246, 246, 246, 0.2);
@location-item-bg-hover: RGBA(246, 246, 246);
@location-item-border: RGBA(0, 17, 34, 0.1);
@location-item-name: #001122;
@location-edit-icon: #FFFFFF;

@modal-ok-button-bg: #37B24D;
@modal-ok-button-color: #ffffff;
