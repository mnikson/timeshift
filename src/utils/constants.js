/**
 * @file constants
 * @description Application constants
 */

// Routes
export const ROUTES = {
  HOME: '/',
  LOCATION: '/location/:id'
};

export const APP = {
  SUBTITLE: 'All locations',
  TITLE: 'Acme locations'
};

/**
 * API Methods
 */
export const API_METHODS = {
  DELETE: 'DELETE',
  GET: 'GET',
  PATCH: 'PATCH',
  POST: 'POST',
  PUT: 'PUT'
};
