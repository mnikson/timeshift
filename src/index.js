/* eslint-disable react/jsx-filename-extension */
// Libs
import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import dotenv from 'dotenv';

import config from './config';

// Components
import App from './components/App';

// Globals
import createStore from './globals/store';

import reportWebVitals from './reportWebVitals';

dotenv.config();

const { store } = createStore();

ReactDOM.render(
  <Provider store={ store }>
    <App />
  </Provider>,
  document.getElementById(config.rootElement)
);

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals();
