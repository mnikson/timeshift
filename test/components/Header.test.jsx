/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';

// Components
import { Header } from '../../src/components';

// Utils
import { APP } from '../../src/utils/constants';

describe('Header component', () => {
  describe('render', () => {
    const props = {
      title: 'All locations',
      subtitle: 'Acme locations'
    };

    it('should render title and subtitle on component', () => {
      render(
        <Header { ...props } />
      );

      const title = screen.getByText(APP.TITLE);
      expect(title).toBeInTheDocument();
      const subtitle = screen.getByText(APP.SUBTITLE);
      expect(subtitle).toBeInTheDocument();
    });

    it('should render component', () => {
      const { container } = render(
        <Header { ...props } />
      );

      expect(container).toMatchSnapshot();
    });
  });
});
