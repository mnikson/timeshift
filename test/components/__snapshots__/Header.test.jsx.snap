// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render should render component 1`] = `
<div>
  <header
    class="ant-layout-header"
  >
    <div>
      All locations
    </div>
    <div>
      Acme locations
    </div>
  </header>
</div>
`;
